src/interop/js-interop/usage.md

```diff
@@ -0,0 +1,454 @@
+    `Options(a: 0)` would result in `{a: 0}`. You can get or set the properties
+    of the resulting object through `external` instance members.
+
+    {{site.alert.warn}}
+    There's a bug that currently requires object literal constructors to have an
+    [`@JS`](#js) annotation on the library, empty or otherwise. See [#54801] for
+    more details.
+    {{site.alert.end}}
+
+- **`static` members**. Like constructors, these members use the name of the
+  extension type to generate the JS code. For example, calling
+  `Time.getTimeDifference(t1, t2)` will generate a JS invocation that looks like
+  `Time.getTimeDifference(t1, t2)`. Similarly, calling `Time.dinnerTime` will
+  result in a JS invocation that looks like `Time.dinnerTime`. Like top-levels,
+  you can declare `static` methods, getters, setters, and fields.
+
+- **Instance members**. Like with other Dart types, these members require an
+  instance in order to be used. These members get, set, or invoke properties on
+  the instance. For example:
+
+  ```dart
+    final time = Time(0, 0);
+    print(time.isDinnerTime()); // false
+    final dinnerTime = Time.dinnerTime;
+    time.hours = dinnerTime.hours;
+    time.minutes = dinnerTime.minutes;
+    print(time.isDinnerTime()); // true
+  ```
+
+  The call to `dinnerTime.hours` gets the value of the `hours` property of
+  `dinnerTime`. Similarly, the call to `time.minutes=` sets the value of the
+  `minutes` property of time. The call to `time.isDinnerTime()` calls the
+  function in the `isDinnerTime` property of `time` and returns the value.
+  Like top-levels and `static` members, you can declare instance methods,
+  getters, setters, and fields.
+
+- **Operators**. There are only two `external` interop operators allowed in
+  interop types: `[]` and `[]=`. These are instance members that match the
+  semantics of JS' [property accessors]. For example, you can declare them like:
+
+  ```dart
+  extension type Array(JSArray<JSNumber> _) implements JSArray<JSNumber> {
+    external JSNumber operator [](int index);
+    external void operator []=(int index, JSNumber value);
+  }
+  ```
+
+  Calling `array[i]` gets the value in the `i`th slot of `array`, and
+  `array[i]=` sets the value in that slot to `i.toJS`. Other JS operators are
+  exposed through [utility functions] in `dart:js_interop`.
+
+Lastly, like any other extension type, you're allowed to declare any
+[non-`external` members] in the interop type. `isMidnight` is one such example.
+
+### Extension members on interop types
+
+You can also write `external` members in [extensions] of interop types. For
+example:
+
+```dart
+extension on Array {
+  external int push(JSAny? any);
+}
+```
+
+The semantics of calling `push` are identical to what it would have been if it
+was in the definition of `Array` instead. The only `external` members you can
+write in such an extension are instance members and the allowed operators. Like
```

Comment:
It may not be clear what cannot be done. Consider:

```suggestion
was in the definition of `Array` instead. Extensions can have `external` instance members and operators, but cannot have `external` static members or constructors. Like
```

Reply:
Done.